### PR TITLE
🔧 feat: `deleteRagFile` utility for Consistent RAG API document deletion

### PR DESCRIPTION
**Pull request by ethanlaj**
# Pull Request Template

## Summary

This pull request refactors and centralizes the logic for deleting embedded files from the RAG API across all file storage backends (S3, Azure, Firebase, Local). It introduces a new shared utility function, `deleteRagFile`, to handle RAG document deletions, replacing duplicated code in each backend. Comprehensive unit tests for this utility are also added.

**Centralization and Refactoring of RAG Deletion Logic:**

* Introduced a new utility function `deleteRagFile` in `packages/api/src/files/rag.ts` to handle deletion of embedded documents from the RAG API, consolidating logic previously duplicated in each storage backend.
* Updated all file storage backend CRUD modules (`Azure`, `Firebase`, `Local`, `S3`) to use the new `deleteRagFile` utility instead of inline RAG deletion logic, simplifying code and ensuring consistent behavior. [[1]](diffhunk://#diff-50720c98138063e48ffb616796f4e09e6edc255943cbc1bc516faeb7590b36e8L7-R7) [[2]](diffhunk://#diff-50720c98138063e48ffb616796f4e09e6edc255943cbc1bc516faeb7590b36e8R105-R107) [[3]](diffhunk://#diff-e0d20f7d56d5674ef4605919416999b1e2dc3d07c2d6d2e5dd57e1152bb6ae98L6-R6) [[4]](diffhunk://#diff-e0d20f7d56d5674ef4605919416999b1e2dc3d07c2d6d2e5dd57e1152bb6ae98L170-R171) [[5]](diffhunk://#diff-8be70ea7d6cc188896e06c7b427cbd92afd1de3e3a9ce4494fb9935d93eac2cfL6-R6) [[6]](diffhunk://#diff-8be70ea7d6cc188896e06c7b427cbd92afd1de3e3a9ce4494fb9935d93eac2cfL211-R212) [[7]](diffhunk://#diff-18be5d015a50a00ae5dbedd67d60517c1a0a7b252633f408c71361855f7a8016L3-R3) [[8]](diffhunk://#diff-18be5d015a50a00ae5dbedd67d60517c1a0a7b252633f408c71361855f7a8016R145-R147)

**Testing and Exports:**

* Added comprehensive unit tests for `deleteRagFile` in `packages/api/src/files/rag.spec.ts`, covering successful deletion, error handling, and edge cases.
* Exported the new `rag` utility module in the files index for broader availability.

## Change Type

Please delete any irrelevant options.

- [X] New feature (non-breaking change which adds functionality)

## Testing

Uploaded a PDF in chat, saw it upload to S3 and the embeddings in PG Vector DB. Then I deleted it in two ways:

1) Without starting the chat (essentially upload and immediately delete) - saw it get removed from S3 and embeddings DB
2) In the "Manage Files" modal - saw it get removed from S3 and embeddings DB

## Checklist

Please delete any irrelevant options.

- [X] My code adheres to this project's style guidelines
- [X] I have performed a self-review of my own code
- [X] I have commented in any complex areas of my code
- [X] My changes do not introduce new warnings
- [X] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes

### CODE REVIEW

File: api/server/services/Files/Firebase/crud.js [hunk 1]
```diff
@@ -167,27 +167,8 @@ function extractFirebaseFilePath(urlString) {
...
-      if (error.response?.status === 404) {
-        logger.warn(
-          `[deleteFirebaseFile] Document ${file.file_id} not found in RAG API, may have been deleted already`,
-        );
-      } else {
-        logger.error('[deleteFirebaseFile] Error deleting document from RAG API:', error);
-      }
-    }
-  }
+  // Delete from RAG API if the file has embeddings
```
Comment (danny-avila) [inline api/server/services/Files/Firebase/crud.js]: remove these extraneous comments please @ethanlaj same comment in a few other places

**Review by danny-avila — CHANGES_REQUESTED** [review]

File: api/server/services/Files/S3/crud.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 const fs = require('fs');
 const fetch = require('node-fetch');
-const { initializeS3 } = require('@librechat/api');
+const { initializeS3, deleteRagFile } = require('@librechat/api');
```
Comment (danny-avila) [inline api/server/services/Files/S3/crud.js]: import order, for package imports, should be imported from the top, shortest to longest line length

File: packages/api/src/files/rag.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	if (!file.embedded || !process.env.RAG_API_URL) {
+		return true;
+	}
+
+	if (!userId) {
+		logger.error('[deleteRagFile] No user ID provided');
+		return false;
+	}
+
+	// Generate a short-lived token for authentication
```
Comment (danny-avila) [inline packages/api/src/files/rag.ts]: please do your best that LLM's don't leave these kinds of comments in the codebase. code should be self-documenting unless something complex is being described

**Review by danny-avila — APPROVED** [review]

**Comment by danny-avila** [comment]
ty for the PR!
